### DISCUSSION TITLE
WalletConnect React & wagmi + Web3js

### DISCUSSION THREAD

**Discussion by 2facedsquid** [WalletConnect/walletconnect-monorepo#1993] [Developer Support]
Hello I started using the React + wagmi web3modal V2 and the question is how to get the Provider object from web3modal to instantiate the Web3 object from the Web3js library, before wallet connect I was using the window.ethereum but that implementation stopped working after importing the WalletConnect packages.

**Comment by br0wnD3v** [comment]
Did you figure it out? Stumbled on the same thing

**Comment by paintoshi** [comment]
I tried a lot of different things a while back, wrote about it here: https://github.com/wagmi-dev/wagmi/discussions/1733

But I could never get it to work reliably when connecting a mobile wallet over WalletConnect. Some transactions would fail even when using the wagmi signer. So we are about to migrate our whole project from web3.js using ethers and wagmi functions instead. It's a lot of work and we are still battling..

Simple answer: WalletConnect and Web3.js are not compatible.
